lib/AppInfo/Application.php

```diff
@@ -0,0 +1,47 @@
+<?php
+namespace OCA\Analysis_app\AppInfo;
+use \OCP\AppFramework\App;
+use OCP\IContainer;
+
+class Application extends App{
+    //Define your dependencies here
+    public function __construct(array $urlParams = array())
+    {
+        parent::__construct('analysis_app', $urlParams);
+
+        $container = $this->getContainer();
+
+        $container->registerService(
+            'URLGenerator', function(IContainer $c) {
+            /** @var \OC\Server $server */
+                $server = $c->query('ServerContainer');
+
+                 return $server->getURLGenerator();
+            }
+        );
+
+        $container->registerService(
+            'UserId', function(IContainer $c) {
+              $user = \OC::$server->getUserSession()
+                   ->getUser();

```

Comment:
indent

Reply:
has been done with spacebar, automatic indentation wouldn't do it properly.

Reply:
?